### DISCUSSION TITLE
Is there any way to pass data as a formula and it gets evaluated automatically

### DISCUSSION THREAD

**Discussion by mohit-11-dev** [iddan/react-spreadsheet#325] [Q&A]
Hi, 
I am trying to implement top formula bar which goes hand in hand with cell just like google sheets
![image](https://github.com/iddan/react-spreadsheet/assets/51900532/484d027f-aab8-487c-bc13-72acb0a144d0)

Steps:  
- pass data as   const data = [
    [{ value: "1" }, { value: "4" }],
    [{ value: "=SUM(A1+B1)" }, { value: "3" }],
  ];
- it should automatically display sum of 1+4 and on data edit mode should show formula similar to how it works in we provide this formula on cell edit but it 
- I am updating data on a formula bar input box change where user could pass a formula but I am seeing only formula string if I send data as this formula string.

**Comment by iddan** [comment]
This is an interesting use case. So you would need some way to get the evaluated formula cell from the spreadsheet?

**Reply by mohit-11-dev** [reply]
I want to pass formula as input and on data load it should get evaluated. 

Behaviour should be similar to how it works when user type formula in a cell and in view mode it shows evaluated value and on edit mode it shows formula

**Reply by mohit-11-dev** [reply]
For example let's say I provide data prop as [
[{ value: "1" }, { value: "4" }],
[{ value: "=SUM(A1+B1)" }, { value: "3" }],
];

It doesn't evaluate formula instead it shows formula text inside cell. 

Same input text if user add inside cell on runtime it show evaluated value on runtime in view mode and on edit mode it shows formula.

So is there any way to replicate same by passing above data as input prop?
